spec/controllers/sites/permissions_controller_spec.rb

```diff
@@ -0,0 +1,82 @@
+require 'rails_helper'
+
+describe Sites::PermissionsController, type: :unit do
+	let(:controller) { described_class.new }
+	let(:params) {
+		ActionController::Parameters.new(
+			site_slug: site.slug
+		)
+	}
+	let(:site) { FactoryBot.create(:site) }
+	let(:request_double) { instance_double('ActionDispatch::Request') }
+	before do
+		allow(request_double).to receive(:host).and_return('localhost')
+		allow(request_double).to receive(:optional_port)
+		allow(request_double).to receive(:protocol)
+		allow(request_double).to receive(:path_parameters).and_return({})
+		allow(request_double).to receive(:flash).and_return({})
+		allow(controller).to receive(:params).and_return(params)
+		allow(controller).to receive(:request).and_return(request_double)
+		allow(controller).to receive(:load_subsite).and_return(site)
+		controller.instance_variable_set(:@subsite, site)
+	end
+	describe '#site_params' do
+		let(:is_admin) { false }
+		before do
+			controller.instance_variable_set(:@subsite, site)
+			allow(controller).to receive(:can?).with(:admin, site).and_return(is_admin)
+		end
+		context 'with editor_uids text' do
+			let(:uids) { ['abc', 'bcd', 'cde'] }
+			let(:params) {
+				ActionController::Parameters.new(
+					site: {
+						editor_uids: uids.join("\n ,")
```

Comment:
```suggestion
						editor_uids: uids.join(",\n")
```
Did you mean to do that?  Or maybe it's just not the format I was expecting.

Reply:
it's intentionally janky to test the array parsing I think

Reply:
I like the idea of testing that, but the testing going on here seems to be less about parsing unexpected variations in input and more about testing updates with/without admin.  Maybe the janky format testing should be a separate thing?

Reply:
Good point, will do.

Reply:
Now in separate specs.